core/src/autogluon/core/models/greedy_ensemble/cmaes_weights.py

```diff
@@ -0,0 +1,448 @@
+from __future__ import annotations
+
+import logging
+import time
+from collections import Counter
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+from ...constants import PROBLEM_TYPES
+from ...metrics import log_loss
+from ...utils import compute_weighted_metric, get_pred_from_proba
+
+# The following import statements are specifically for the CMAESWeights class 
+import cma
+from functools import partial
+from math import isclose
+# End of the import statements 
+
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractWeightedEnsemble:
+    def predict(self, X):
+        y_pred_proba = self.predict_proba(X)
+        return get_pred_from_proba(y_pred_proba=y_pred_proba, problem_type=self.problem_type)
+
+    def predict_proba(self, X):
+        return self.weight_pred_probas(X, weights=self.weights_)
+
+    @staticmethod
+    def weight_pred_probas(pred_probas, weights):
+        """
+        This method takes predicted probabilities from multiple models and a list of weights corresponding to those models, 
+        and then calculates a weighted ensemble prediction. 
+
+        Parameters
+        ----------
+        pred_probas: np.ndarray
+            Array of prediction probabilities from multiple models
+        weights: np.ndarray
+            Weightages corresponding to different models 
+
+        Returns
+        -------
+        preds_ensemble: np.ndarray
+            Weighted ensemble prediction given the passed in predictions probabilities and weights
+        """
+
+        preds_norm = [pred * weight for pred, weight in zip(pred_probas, weights)]
+        preds_ensemble = np.sum(preds_norm, axis=0)
+        return preds_ensemble
```

Comment:
Why are we duplicating code that exists in https://github.com/alee4977/autogluon/blob/731e61a7be08e2f7dc8b2aa81873d649ff165cb2/core/src/autogluon/core/models/greedy_ensemble/ensemble_selection.py ?